atom/browser/native_window_mac.mm

```diff
@@ -1100,8 +1100,15 @@ void ViewDidMoveToSuperview(NSView* self, SEL _cmd) {
 void NativeWindowMac::SetOverlayIcon(const gfx::Image& overlay,
                                      const std::string& description) {}
 
-void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible) {
-  SetCollectionBehavior(visible, NSWindowCollectionBehaviorCanJoinAllSpaces);
+void NativeWindowMac::SetVisibleOnAllWorkspaces(bool visible,
+                                                bool visibleOnFullScreen) {
+  if (visible && visibleOnFullScreen) {
+    SetCollectionBehavior(visible,
```

Comment:
I feel as though there are going to be edge cases to this masking logic, can we make the masking of `NSWindowCollectionBehaviorCanJoinAllSpaces ` independent of the masking for `NSWindowCollectionBehaviorFullScreenAuxiliary` otherwise I think users can get into an unexpected state by calling 

```js
SetVisibleOnAllWorkspaces(true, true);
SetVisibleOnAllWorkspaces(false);
SetVisibleOnAllWorkspaces(true, false); // This would appear to be the result of true, true again
```

Reply:
Thanks for the quick reply!

I just went ahead and removed the if/else and that should take care of all cases